packages/fuels-contract/src/script_calls.rs

```diff
@@ -0,0 +1,181 @@
+use std::fmt::Debug;
+use std::marker::PhantomData;
+
+use fuel_gql_client::fuel_tx::{Output, Receipt, Transaction};
+use fuel_tx::Input;
+
+use crate::contract::get_decoded_output;
+
+use fuels_core::{
+    parameters::{CallParameters, TxParameters},
+    Tokenizable,
+};
+use fuels_signers::{provider::Provider, WalletUnlocked};
+
+use fuels_types::{errors::Error, param_types::ParamType};
+
+use crate::execution_script::{CompiledScript, TransactionExecution};
+
+pub struct ScriptInterface {
+    pub script: TransactionExecution,
+    pub wallet: WalletUnlocked,
+}
+
+#[derive(Debug)]
+pub struct ScriptCallResponse<D> {
```

Comment:
This seems to implement the same behaviour as `ContractCallResponse`. Keeping both is more mental strain for the user and cumbersome to maintain. 
How about generalising this to something like `VMResponse`

Another thing to keep in mind is that #690 moves log methods to the response. ATM I'm unsure of what implication logging has in the context of scripts.